scripts/analysis.py

```diff
@@ -0,0 +1,378 @@
+    sum = 0
+    for ar in array:
+        sum += ar[column_index]
+
+    return sum
+
+
+
+def isotope_calc(wasteid_array, snf_inventory, cursor):
+    """ Calculates isotope mass using mass fraction in compositions table.
+
+        Fetches all from compositions table.
+        Compositions table has the following format:
+            SimId / QualId / NucId / MassFrac
+        Then sees if the qualid matches, and if it multiplies
+        the mass fraction by the snf_inventory.
+
+    Prameters
+    ---------
+    wasteid_array: array
+        array of qualid of wastes
+    snf_inventory: float
+        total mass of snf
+    cursor: cursor
+        cursor for sqlite3
+
+    Returns
+    -------
+    nuclide_inven: array
+        inventory of individual nuclides.
+    """
+
+    # Get compositions of different commodities
+    # SimId / QualId / NucId / MassFrac
+    cur = cursor
+    comp = cur.execute('select * from compositions').fetchall()
+    total_snf_mass = get_sum(snf_inventory, 0)
+
+    nuclide_inven = 'total snf inventory = ' + str(total_snf_mass) + 'kg \n'
+    nuclides = []
+    mass_of_nuclides = []
+    # if the 'qualid's match,
+    # the nuclide quantity and calculated and displayed.
+    for isotope in comp:
+        for num in snf_inventory:
+            # num[1] = snf inventory qualid 
+            # isotope[1] = compositions qualid 
+            if num[1] == isotope[1]:
+                # num[0] = total mass of one composition
+                # isotope[3] = mass fraction
+                nuclide_quantity = num[0] * isotope[3]
+                #isotope[2] = nucid
+                nuclide_name = isotope[2]
+                nuclides.append(nuclide_name)
+                mass_of_nuclides.append(nuclide_quantity)
+
+    nuclide_set = set(nuclides)
+    mass_dict = collections.OrderedDict({})
+
+    for nuclide in nuclide_set:
+        temp_nuclide_sum = 0 
+        for i in range(len(nuclides)):
+            if nuclides[i] == nuclide:
+                temp_nuclide_sum += mass_of_nuclides[i]
+        mass_dict[nuclide] = temp_nuclide_sum
+
+
+    for nuclide in mass_dict:
+        nuclide_name = str(nucname.name(nuclide))
+        nuclide_inven += nuclide_name + ' = ' + str(mass_dict[nuclide]) + 'kg \n'
+
+    return nuclide_inven
+
+
+def capacity_calc(governments, timestep, entry, exit):
+    """ Adds and subtracts capacity over time for plotting
+
+    Paramters
+    ---------
+    governments: array
+        array of governments (countries)
+    timestep: array
+        array of timestep from 0 to simulation time
+    entry: array
+        power_cap, agentid, parentid, entertime
+        of all entered reactors
+    exit: array
+        power_cap, agentid, parenitd, exittime
+        of all decommissioned reactors
+
+    Returns
+    -------
+    power_dict: dictionary
+        dictionary of capacity progression with country_government as key
+    num_dict: dictionary
+        dictionary of number of reactors progression
```

Comment:
number of reactors progression sounds odd. I thinl you mean "timeseries of the number of reactors operating"